sigstore/_internal/fulcio/client.py

```diff
@@ -262,14 +267,14 @@ def _opaque16(value: bytes) -> bytes:
             raw_sct = _opaque16(_opaque16(raw_sct_list_bytes))
 
             sct = precert_scts_extension[0]
-        except ExtensionNotFound:
+        else:
             # If we don't have any embedded SCTs, then we might be dealing
             # with a Fulcio instance that provides detached SCTs.
 
             # The SCT header is a base64-encoded payload, which in turn
             # is a JSON representation of the SignedCertificateTimestamp
             # in RFC 6962 (subsec. 3.2).
-            sct_b64 = resp.headers.get("SCT")
+            sct_b64 = resp.json()["signedCertificateDetachedSct"].get("signedCertificateTimestamp")
```

Comment:
We probably need to handle the case where `signedCertificateDetachedSct` is missing as well.

Reply:
Do we need to handle this case? That would be a malformed response from the server if it was missing.

Reply:
Same question for `resp.json()["signedCertificateEmbeddedSct"]["chain"]["certificates"]` - Do we need to guard against a malformed response, or are we ok with the key error that's returned?

Reply:
IMO yes, we should wrap this whole thing in a `try`/`except KeyError` and produce a more meaningful error message than the stacktrace that would appear otherwise.

Reply:
Done! It's been awhile since I've written Python, forgot about the power of `try/except`.